src/backend/storage/lmgr/lmgr.c

```diff
@@ -332,6 +333,9 @@ CheckRelationLockedByMe(Relation relation, LOCKMODE lockmode, bool orstronger)
 {
 	LOCKTAG		tag;
 
+	if (get_ENR_withoid(currentQueryEnv, RelationGetRelid(relation), ENR_TSQL_TEMP))
```

Comment:
can we wrap `get_ENR_withoid(currentQueryEnv, [relid], ENR_TSQL_TEMP)` pattern in a macro, since it is used in many temp table related places. would improve readability